### ISSUE TITLE
Console error with symfony 6.4

### ISSUE THREAD

**Issue by jurv** [Sylius/Sylius#15575]
**Sylius version affected**: 1.12

**Description**  

The newly release Symfony 6.4.0 is now the version used when creating a new Sylius 1.12 project, but the console is not usable after project creation. The `cache:clear` command fails at the end of the `composer create-project`, and after that every call to `bin/console` produce this error:

> An exception occurred in the driver: SQLSTATE[HY000] [1698] Access denied for user 'root'@'localhost' 

The issue is related to the instanciation of `Sylius\Bundle\ChannelBundle\Collector\ChannelCollector` in the `sortListeners` method of `Symfony\Component\EventDispatcher\EventDispatcher`:

```php
$listener[0] = $listener[0]();
```

The constructor of `ChannelCollector` contains a call to a repository method,  which fails because of missing/unconfigured database:
```php
$channelRepository->findAll();
```

 

**Steps to reproduce**  

```bash
composer create-project sylius/sylius-standard bug-reproduction-project
```

**Possible Solution**  

If my case (automated test of sylius plugin), I restricted the symfony version in composer to prevent using symfony 6.4:

```bash
# Create the project but don't install anything
composer create-project --no-interaction --prefer-dist --no-scripts --no-progress --no-install sylius/sylius-standard="~1.12.0" .

# Before installation, add a constraint on symfony version
composer config extra.symfony.require "6.3.*"

# Now that symfony 6.4 can't be installed, proceed with installation
composer install --no-interaction
```

**Comment by Lionel-D** [comment]
Reproduced, hoping for a quick fix

**Comment by maximehuran** [comment]
Same on Monsieur Biz' plugins

**Comment by maximehuran** [comment]
@jurv Does it work on the `install` when you force the version ?

Because at the install it is not loaded. 
I already had this issue before : https://github.com/composer/composer/issues/10338#issuecomment-988697300

**Comment by GSadee** [comment]
This issue should be fixed by #15580

**Comment by jurv** [comment]
> @jurv Does it work on the `install` when you force the version ?
> 
> Because at the install it is not loaded. I already had this issue before : [composer/composer#10338 (comment)](https://github.com/composer/composer/issues/10338#issuecomment-988697300)

@maximehuran I manage to make it work in my Makefile (widely inspired by yours BTW :wink:) by doing something like this:

```Makefile
setup_application: ${APP_DIR}/.php-version ${APP_DIR}/.env.local
	rm -f ${APP_DIR}/yarn.lock
	(cd ${APP_DIR} && ${COMPOSER} config repositories.core '{"type": "composer", "url": "[...]"}')
	(cd ${APP_DIR} && ${COMPOSER} config repositories.plugin '{"type": "path", "url": "../../"}')
	(cd ${APP_DIR} && ${COMPOSER} config extra.symfony.allow-contrib true)
	(cd ${APP_DIR} && ${COMPOSER} config minimum-stability dev)

	# Prevent installing symfony 6.4 for now as cache:clear fails because of missing database
ifeq ("${SYLIUS_VERSION}", "1.12")
	(cd ${APP_DIR} && ${COMPOSER} config extra.symfony.require "6.3.*")
endif

	(cd ${APP_DIR} && ${COMPOSER} config --no-plugins allow-plugins true)
	(cd ${APP_DIR} && ${COMPOSER} install --no-interaction)
```

@GSadee thanks, indeed it looks like it will fix it. As this is a annoying issue preventing proper installation of projects, is a new release soon to be published with this fix ?
